### PR TITLE
[JIT] Revert  Freezing shared type PR

### LINKED ISSUE #45902: Freeze module pass fails on torchvision RCNN models
## 🐛 Bug

Freeze module pass fails with below exception for torchvision Faster RCNN, Mask RCNN, Keypoint RCNN models:

```
torch._C._freeze_module(script_module._c)
RuntimeError: node->kind() == prim::GetAttr INTERNAL ASSERT FAILED at "../torch/csrc/jit/passes/freeze_module.cpp":22, please report a bug to PyTorch. Expected prim::GetAttr nodes
```
## To Reproduce

```
model = torchvision.models.detection.keypoint_rcnn.keypointrcnn_resnet50_fpn(pretrained=True, min_size=200, max_size=300)
model.eval()
script_module = torch.jit.script(model)
torch._C._freeze_module(script_module._c)
```

cc @gmagogsfm

### PR DESCRIPTION
**Pull request by eellison**
Fixes https://github.com/pytorch/pytorch/issues/45902 by reverting https://github.com/pytorch/pytorch/pull/42457 

The test case introduced by https://github.com/pytorch/pytorch/pull/42457 was fixed by https://github.com/pytorch/pytorch/pull/46250, which I'm assuming is the real source of the bug.

In the future it would be good to provide repro's for freezing issues without including a quantization dependency; there was another another issue in freezing (see: https://github.com/pytorch/pytorch/pull/46054) who's root cause was the same quantization issue https://github.com/pytorch/pytorch/pull/46250.

### CODE REVIEW

**Review by vkuzo — APPROVED** [review]
accepting to unblock, please feel free to wait for @bzinodev for a more proper review if needed (I do not know this area of the code well)

**Comment by eellison** [comment]
This is failing indeterminately for me now, and it always passes when the test is run alone. :/ 
```
(base) eellison@devfair044:~/pytorch$ python test/test_jit.py TestFreezing
Fail to import hypothesis in common_utils, tests are not derandomized
.................................../private/home/eellison/miniconda3/lib/python3.7/site-packages/torch/quantization/observer.py:123: UserWarning: Please use quant_min and quant_max to specify the range for observers.                     reduce_range will be deprecated in a future release of PyTorch.
  reduce_range will be deprecated in a future release of PyTorch."
.
----------------------------------------------------------------------
Ran 36 tests in 3.371s

OK
(base) eellison@devfair044:~/pytorch$ python test/test_jit.py TestFreezing
Fail to import hypothesis in common_utils, tests are not derandomized
.................................../private/home/eellison/miniconda3/lib/python3.7/site-packages/torch/quantization/observer.py:123: UserWarning: Please use quant_min and quant_max to specify the range for observers.                     reduce_range will be deprecated in a future release of PyTorch.
  reduce_range will be deprecated in a future release of PyTorch." 
```

@kimishpatel any insight on where the original failure came from ? This PR does regress that test case but the original PR regressed freezing so I'm not sure how we should continue but we should at least consider landing this

**Comment by kimishpatel** [comment]
I can try taking a look today.

**Comment by eellison** [comment]
@kimishpatel i think i have a better handle on the issue, i updated the patch. Will wait for you or @bzinodev to approve.

File: torch/csrc/jit/passes/freeze_module.cpp
Comment (kimishpatel) [inline torch/csrc/jit/passes/freeze_module.cpp:569]: I dont quite follow why not breaking from loop helps. Does it mean that you gonna now add modules you get from 'getattr' nodes to the list and always iter over all of them so as not to not skip any of them. And then for a getattr node also match against all possible instances of the module type?

Comment (eellison) [reply torch/csrc/jit/passes/freeze_module.cpp:569]: Yes, it means for getAttr we're matching against all possible instances of the module type, instead of just the first one (first version) or the one that the gettAttr chain resolves to bc:
> we would have to be careful that the chain does not contain any attributes which get written to (setAttr calls)

Comment (eellison) [reply torch/csrc/jit/passes/freeze_module.cpp:569]: We could try to be more exact and follow getAttrs calls + make sure that none of those attributes were mutable, but that should be a follow up, we should get correctness down + make freezing not fail on models first

Comment (kimishpatel) [reply torch/csrc/jit/passes/freeze_module.cpp:569]: Yeah that sounds good. In few instances of reported breakage of freeze_module because of this PR was generally related to this issue of not being able follow chain of getattrs. Thanks for the test case. I kind of see that now.

**Pull request closed without merging**
